### PR TITLE
SSCSCI-529

### PR DESCRIPTION
**Pull request by hashimalisolirius**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCSCI-529

### Change description ###
Enable hearings on preview

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by NHAL22 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java [hunk 1]
```diff
@@ -18,24 +18,28 @@
 
 import static org.springframework.http.HttpHeaders.AUTHORIZATION;
 
+@SuppressWarnings({"PMD.UseObjectForClearerAPI"})
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java:21]: what is triggering this?

File: src/main/java/uk/gov/hmcts/reform/sscs/jms/listener/HmcHearingsEventTopicListener.java [hunk 2]
```diff
@@ -65,23 +66,30 @@ public void onMessage(JmsBytesMessage message) throws JMSException, HmcEventProc
...
+
     }
 
     private boolean isMessageRelevantForService(HmcMessage hmcMessage) {
         return sscsServiceCode.equals(hmcMessage.getHmctsServiceCode());
     }
 
     private boolean isMessageReleventForDeployment(JmsBytesMessage message) throws JMSException {
-        return hmctsDeploymentId == null
+        return hmctsDeploymentId.isEmpty()
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/sscs/jms/listener/HmcHearingsEventTopicListener.java:92]: are all the different permutations of this method covered in the test?

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java [hunk 3]
```diff
@@ -62,6 +62,17 @@ public class HearingsService {
...
     private static final Long HEARING_VERSION_NUMBER = 1L;
 
+    @Value("${idam.oauth2.user.email}")
+    private String idamOauth2UserEmail;
+
+    @Value("${idam.oauth2.user.password}")
+    private String idamOauth2UserPassword;
+
+    @Value("${idam.client.id}")
+    private String idamClientId;
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java:72]: where is the code that uses these new fields?

Comment (hashimalisolirius) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsService.java:71]: Unneeded fields used for logging during testing,  will remove.

File: src/main/java/uk/gov/hmcts/reform/sscs/jms/listener/HmcHearingsEventTopicListener.java (hunk 2 above)
Comment (hashimalisolirius) [reply src/main/java/uk/gov/hmcts/reform/sscs/jms/listener/HmcHearingsEventTopicListener.java:92]: Have added two more tests to cover the possible scenarios, now covers:
- message and service id match
- message and service id dont match
- empty service id and empty message id 
- empty service id but message id

**Review by benouaer — DISMISSED** [review]

**Review by benouaer — DISMISSED** [review]

**Review by benouaer — APPROVED** [review]

**Review by KatKovacs1 — APPROVED** [review]

**Pull request merged**